### PR TITLE
nixos/mediatomb: wait for network-online.target + fix test

### PR DESCRIPTION
**Pull request by B4dM4n**
###### Motivation for this change

Gerbera might fail if the network interface is not available on startup, so wait for `network-online.target`.

This also fixes the test for Mediatomb/Gerbera and splits the two test cases into two separate tests for more clarity.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [x] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by ardumont — APPROVED** [review]
lgtm

**Comment by schmittlauch** [comment]
Once this is merged and works well, we might want to consider removing mediatomb alltogether for 22.05 in favour of gerbera, or is there any reason for not doing so?

The deprecation notice was already in 21.05 release notes.

**Comment by alyssais** [comment]
Hi, I'm making an effort to get every NixOS test to pass, and the next one on my list is mediatomb.  Could you update this PR to drop the mediatomb test, and just keep the gerbera one?  (Mediatomb has now been entirely removed from Nixpkgs.)

File: nixos/modules/services/misc/mediatomb.nix [hunk 1]
```diff
@@ -363,7 +363,9 @@ in {
     in mkIf cfg.enable {
     systemd.services.mediatomb = {
       description = "${cfg.serverName} media Server";
-      after = [ "network.target" ];
+      # Gerbera might fail if the network interface is not available on startup, so use network-online.target
+      # https://github.com/gerbera/gerbera/issues/1324
+      after = [ "network-online.target" ];
```
Comment (SuperSandro2000) [inline nixos/modules/services/misc/mediatomb.nix:368]:
```suggestion
      # Gerbera might fail if the network interface is not available on startup
      # https://github.com/gerbera/gerbera/issues/1324
      after = [ "network.target" "network-online.target" ];
```

**Comment by B4dM4n** [comment]
The mediatomb test is now removed.

I was also able to remove the "hostnames are not supported" section, since gerbera has been updated in the meantime.

**Comment by SuperSandro2000** [comment]
@ofborg test mediatomb

**Pull request merged**